app/src/main/java/com/github/sdp/mediato/ui/SearchFragment.java

```diff
@@ -147,35 +145,37 @@ public void onClick(View view) {
         this.currentHighlightedButton.setPaintFlags(this.currentHighlightedButton.getPaintFlags() | Paint.UNDERLINE_TEXT_FLAG);
     }
 
-    private void SwitchGridOrRecyclerView(View view) {
-        if (view == peopleButton) {
-            this.gridView.setVisibility(View.GONE);
-            this.recyclerView.setVisibility(View.VISIBLE);
+    private void searchAndDisplayResult(String toBeSearched) {
+        if (this.currentCategory == SearchCategory.PEOPLE) {
+            recyclerView.setLayoutManager(new LinearLayoutManager(this.getContext()));
+            recyclerView.setAdapter(new UserAdapter(searchUserViewModel));
+            searchUser(toBeSearched);
         } else {
-            this.recyclerView.setVisibility(View.GONE);
-            this.gridView.setVisibility(View.VISIBLE);
+           searchMedia(toBeSearched);
         }
     }
 
-    private List<Media> search(String toBeSearched) {
-        List<Media> searchMediaResults = new ArrayList<Media>();
+    private void searchMedia(String toBeSearched){
         switch (this.currentCategory) {
-            case PEOPLE:
-                searchUser(toBeSearched);
-                break;
             case MOVIES:
-                // toDO : fetch from firebase 
-                // for now fetches from a local database
-                if (toBeSearched.equals("james bond")) {
-                    searchMediaResults = new LocalFilmDatabase().getMovieItems();
+                if (!toBeSearched.isEmpty()) {
+                    // fetch from API
+                    theMovieDB.searchItems(toBeSearched, 40).thenAccept(list -> {
+                        searchMediaResults.setValue(list.stream().map(Movie::new).collect(Collectors.toList()));
+                    });
+                } else {
+                    searchMediaResults.setValue(Collections.emptyList());
                 }
                 break;
             case BOOKS:
                 break;
             case MUSIC:
                 break;
         }
-        return searchMediaResults;
+        recyclerView.setLayoutManager(new GridLayoutManager(getContext(), 2));
```

Comment:
This function is completely disconnected from the UI and should be put in a view Model (if possible).